File: Bit Manipulation/Bit-Sum.cpp
int solve(std::vector<int>& nums, int k) {
  std::vector<int> bits_frequency(30);
  for (int num : nums) {
    for (int bit = 0; bit < 30; ++bit) {
      if (num & (1 << bit)) {
        ++bits_frequency[bit];
      }
    }
  }

  const int length = nums.size(); 
  for (int bit = 0; bit < 30; ++bit) {
    const int off_bits = length - bits_frequency[bit];
    const int change = std::min(off_bits, k);
    bits_frequency[bit] += change;
    k -= change;
  }

  int64_t sum = 0;
  for (int bit = 0; bit < 30; ++bit) {
    sum += (int64_t)bits_frequency[bit] << bit;
  }
  return sum % 1'000'000'007;
}

File: Bit Manipulation/Bit-Sum.py
class Solution:
  def solve(self, nums, k):
    bits_frequency = [0 for _ in range(30)]
    for num in nums:
      for bit in range(30):
        if num & (1 << bit):
          bits_frequency[bit] += 1

    for bit in range(30):
      off_bits = len(nums) - bits_frequency[bit]
      change = min(off_bits, k)
      bits_frequency[bit] += change
      k -= change

    return sum(bits_frequency[bit] << bit for bit in range(30)) % (10**9 + 7)
